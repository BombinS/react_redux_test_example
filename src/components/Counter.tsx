import * as React from 'react';

class Counter extends React.Component<any, any>{

  constructor(props: any) {
    super(props);
    this.state = { desc: '' };
  }

  public render() {
    return (
      <div>
        <button className='button1' onClick={this.onIncrement}>+</button>
        <p className='counts'>{this.state.desc}</p>
        <button className='button2' onClick={this.onDecrement}>-</button>
      </div>
    )
  }

  private onIncrement = () => {
    this.props.store.dispatch({ type: 'INCREMENT' });
    this.toChangeState(`Store was incremented.`);
    this.toDoSomething('Doing something');
  }

  private onDecrement = () => {
    this.props.store.dispatch({ type: 'DECREMENT' });
    this.toChangeState(`Store was decremented.`);
  }

  private toChangeState(arg: string) {
    const desc = `${arg} Current value: ${this.props.store.getState()}`;
    this.setState({ desc });
  }

  private toDoSomething = (arg: string) => {
    console.log(arg);
  }
}

export default Counter;