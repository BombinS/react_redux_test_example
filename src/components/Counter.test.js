import * as React from 'react';
import { createStore } from 'redux'
import counter from '../reducers/counter'
import Counter from './Counter'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('`Counter` component testing', () => {

    it('Reducer test', () => {
        // create store
        const store = createStore(counter)
        // initial store should be equal to 0
        expect(store.getState()).toBe(0)
        // increment
        store.dispatch({ type: 'INCREMENT' })
        expect(store.getState()).toBe(1)
        // decrement
        store.dispatch({ type: 'DECREMENT' })
        expect(store.getState()).toBe(0)

    })

    it('Button pressed test', () => {
        // create store
        const store = createStore(counter)
        // initial store should be equal to 0
        expect(store.getState()).toBe(0)

        const wrapper = shallow(<Counter store={store} />);

        // button should exist
        expect(wrapper.find('.button1').length).toBe(1)

        // increment button should store ++
        wrapper.find('.button1').simulate('click')
        expect(store.getState()).toBe(1)

        // decrement button should store --
        wrapper.find('.button2').simulate('click')
        expect(store.getState()).toBe(0)

    })

    it('Button pressed test and stub', () => {
        // create store
        const store = createStore(counter)
        // initial store should be equal to 0
        expect(store.getState()).toBe(0)

        const wrapper = shallow(<Counter store={store} />);
        wrapper.instance().toChangeState = jest.fn();
        wrapper.instance().forceUpdate();

        // button should exist
        expect(wrapper.find('.button1').length).toBe(1)

        // increment button should store ++
        wrapper.find('.button1').simulate('click')
        expect(store.getState()).toBe(1)
        expect(wrapper.instance().toChangeState).toBeCalledWith('Store was incremented.')

        // decrement button should store --
        wrapper.find('.button2').simulate('click')
        expect(store.getState()).toBe(0)

    })

})