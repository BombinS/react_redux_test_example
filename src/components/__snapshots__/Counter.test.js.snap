// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Counter\` component testing snapshot 1`] = `
<div>
  <div>
    <h1>
       Hello 
    </h1>
  </div>
  <div>
    <button
      className="button1"
      onClick={[Function]}
    >
      +
    </button>
    <h1
      className="counts"
    >
      
    </h1>
    <button
      className="button2"
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
